autonomous_software_pkg/src/map_publisher_2.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python3
+
+import os
+import rospy
+import tf
+import tf2_ros
+import utm
+
+from enum import Enum
+from pykml import parser
+
+from geometry_msgs.msg import Point, TransformStamped
+from visualization_msgs.msg import Marker, MarkerArray
+
+
+class PlacemarkType(Enum):
+    Contour = 1
+    Lane = 2
+    Obstacle = 3
+
+
+class MapComponent:
+    def __init__(
+        self,
+        id: int = 0,
+        type: PlacemarkType = PlacemarkType.Contour,
+        list_coordinates: list = [],
+    ):
+        self.id = id
+        self.type = type
+        self.list_coordinates = list_coordinates
+
+
+class MapPublisher:
+    def __init__(self):
+
+        rospy.init_node("map_publisher", anonymous=True)
+
+        # Publisher
+        self.map_marker_pub = rospy.Publisher(
+            "map_viz", MarkerArray, queue_size=10, latch=True
+        )
+
+        # Parameters
+        self.map_file_name = rospy.get_param("~map_file_name", "MapSanMateo.kml")
+
+    def load_map(self):
+        """
+        Function to parse a kml file into a list of MapComponent, each defining a portion of the map
+        Returns:
+            - list_components: List of MapComponent
+        """
+
+        waypoint_file_folder_path = os.path.join(
+            os.path.dirname(
+                os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+            ),
+            "utils/utm_map_generation/kml_files",
+        )
+
+        wp_file = os.path.join(waypoint_file_folder_path, self.map_file_name)
+
+        total_id = 0
+
+        list_components = []
+
+        with open(wp_file) as f:
+
+            tree = parser.parse(f)
+
+            root = tree.getroot()
+
+            for p in root.Document.Placemark:
+
+                component = MapComponent(id=total_id)
+                total_id += 1
```

Comment:
You could use `enumerate` to avoid having to keep track of the index by hand.

Reply:
Done